File: 2646-kth-largest-sum-in-a-binary-tree/2646-kth-largest-sum-in-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    long long kthLargestLevelSum(TreeNode* root, int k) {


        queue<TreeNode*> q;
        q.push(root);
        vector<long long>sums;
        for(int level =1, sz= q.size(); !q.empty(); sz = q.size(), level++)
        {
            long long sum = 0;
            while(sz--)
            {
                TreeNode* cur = q.front();
                q.pop();
                if(cur->left)
                    q.push(cur->left);
                if(cur->right)
                    q.push(cur->right);
                sum+=cur->val;
            }
            sums.push_back(sum);
        }
        if(sums.size()<k)
            return -1;
        sort(sums.rbegin(), sums.rend());
        return sums[k-1];
        
    }
};

File: 2646-kth-largest-sum-in-a-binary-tree/2646-kth-largest-sum-in-a-binary-tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution {
    public long KthLargestLevelSum(TreeNode root, int k) {
        Queue<TreeNode> q = new Queue<TreeNode>();
        q.Enqueue(root);
        List<long>sums = new List<long>();
        long sum = 0;
        for(int sz= q.Count(); q.Count()>0; sz = q.Count())
        {
            while(sz-- > 0)
            {
                TreeNode cur = q.Peek();
                q.Dequeue();
                if(cur.left != null)
                    q.Enqueue(cur.left);
                if(cur.right != null)
                    q.Enqueue(cur.right);
                sum+=cur.val;
            }
            sums.Add(sum);
            sum=0;
        }
        if(sums.Count()<k)
            return -1;
        sums.Sort();
        int x = sums.Count();
        return sums[x-k];
        
    }
}